### PR TITLE
Add HttpServer.BusyThreads.Max metric

### PR DESCRIPTION
**Pull request by johngmyers**
Add HttpServer.BusyThreads.Max metric

### CODE REVIEW

File: http-server/src/main/java/com/proofpoint/http/server/HttpServer.java [hunk 1]
```diff
@@ -119,7 +122,16 @@ public HttpServer(HttpServerInfo httpServerInfo,
         checkNotNull(queryStringFilter, "queryStringFilter is null");
         checkNotNull(theServlet, "theServlet is null");
 
-        QueuedThreadPool threadPool = new QueuedThreadPool(config.getMaxThreads());
+        QueuedThreadPool threadPool = new QueuedThreadPool(config.getMaxThreads()) {
+
+            @Override
+            protected void runJob(Runnable job)
```
Comment (gwittel) [inline http-server/src/main/java/com/proofpoint/http/server/HttpServer.java]: Is there a chance that the +1 and -1 will get split across buckets?  Could that make the min/max look weird under certain conditions?

Comment (johngmyers) [inline http-server/src/main/java/com/proofpoint/http/server/HttpServer.java]: That adjusts the current value of `MaxGauge`. Only the maximums are bucketed.

Comment (gwittel) [inline http-server/src/main/java/com/proofpoint/http/server/HttpServer.java]: Right; makes sense.

**Comment by gwittel** [comment]
LGTM

**Pull request merged**